### ISSUE TITLE
Is possible to Generate Lightmaps without modifying the Blender Scene Materials

### ISSUE THREAD

**Issue by nicoraf** [Naxela/The_Lightmapper#154]
Hello, for a particular pipeline, we are interested to generate the lightmaps of the Meshes, we are interested in baked texture and the UV mapping , but we want to keep the material of the Mesh preserved. Is there a way to achieve that? I guess that if not it will be a nice to have.

**Comment by Naxela** [comment]
Hi and thanks for using the addon! - If I understand you correctly you just want to bake the lightmaps without actually applying them to the materials in your scene? In that case, it's possible using the headless toggle:

![image](https://user-images.githubusercontent.com/5429817/187763787-be82e979-a6ed-48db-a839-6d76a11eaa3e.png)

That should work in case you want to export your scenes to external game engines, using gltf or similar

I hope this helps! - If not let me know

**Comment by nicoraf** [comment]
Hi @Naxela , thanks for your message. I tested it before raising the issue, with the latest version of Blender 3.2.2 . If I Select Don't apply materials, and then I click "Build lightmaps" , I don't see at blender any Image with with a name "MeshName_bake" .
Ideally I would expect to see the image and also the light UV map .
Thanks

**Comment by Naxela** [comment]
Hi @nicoraf - All your baked images should be present in a folder located where your blend file is:

![image](https://user-images.githubusercontent.com/5429817/187766834-10748a30-2f13-4121-83df-ec0948145623.png)

If you bake with headless mode turned on, do you mean that no lightmaps are saved in the lightmap folder - If you click Explore Lightmaps?:

![image](https://user-images.githubusercontent.com/5429817/187766974-934e697f-e8d7-4cba-b686-667910698562.png)

Or do you just want to have the images loaded but not applied inside the blender file itself (Visible in the image editor for example)?

**Comment by nicoraf** [comment]
Yes we want to have the images embedded inside the blender file itself, but not attached to a material.

**Comment by Naxela** [comment]
Oh now I understand - I don't think there's currently a way to do that while in headless mode. I can try to add the feature sometime during the weekend, but until then can you test if the following script work?:

```python
import bpy, os

for file in os.listdir(os.path.join(os.path.dirname(bpy.data.filepath), bpy.context.scene.TLM_EngineProperties.tlm_lightmap_savedir)):
    if file.endswith(".hdr"):
        bpy.ops.image.open(filepath=os.path.join(os.path.join(os.path.dirname(bpy.data.filepath), bpy.context.scene.TLM_EngineProperties.tlm_lightmap_savedir), file))
```

Just copy paste the code into the scripting panel:

![image](https://user-images.githubusercontent.com/5429817/187769487-be3fdd8a-1026-427d-8a40-d94f256c0a95.png)

And your lightmaps should now be loaded:

![image](https://user-images.githubusercontent.com/5429817/187769574-a7b11058-c5e1-476e-90ee-9a26ceb21502.png)

I hope this helps

**Comment by nicoraf** [comment]
Thanks, we will wait you add this feature. We appreciate it.

So I guess the natural way will be to add a combination of check options so that "Build lightmaps", include the bake images embedded with the same name convention, but it does not modify the materials. However the UV mapping is also inside the Mesh objects.

**Comment by Naxela** [comment]
Hi, I added this as an option in the latest commit (https://github.com/Naxela/The_Lightmapper/commit/4bc61625edae4726660b28feb88146b62021a3fc):

After you've baked, just go into utilities, and select load lightmaps at runtime:

![image](https://user-images.githubusercontent.com/5429817/188322920-ba9b86a7-fb16-4692-93d9-b48d2beb931c.png)

**Comment by nicoraf** [comment]
Hi @Naxela , thank you very much , we will be testing it and I will comment here again.

**Comment by nicoraf** [comment]
Hi @Naxela , we tested your changes and we have the following feedback.

1) When we select load lightmaps, it loads the lightmap image textures, but it only loads image textures with HDR format. We are interested to load PNG images and ideally to be able to choose between PNG and JPEG. One reason is because we prefer that format, but there is another reason is that to our understanding the GLTF exporter add on does not support HDR.

2) Another comment that is independent of 1). Although the approach of doing that in two steps work for us. Ideally we would prefer that the add on, directly load the images after backing, without modifying the materials. Maybe it could be added an option "Pack Backed Images", and another combo that we can choose the image format PNG, JPEG, etc ?

Thanks
